mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -272,12 +272,15 @@ class ChiSqSelector @Since("2.1.0") () extends Serializable {
         // https://en.wikipedia.org/wiki/False_discovery_rate#Benjamini.E2.80.93Hochberg_procedure
         val tempRes = chiSqTestResult
           .sortBy { case (res, _) => res.pValue }
-        val maxIndex = tempRes
+        val selected = tempRes
           .zipWithIndex
           .filter { case ((res, _), index) =>
             res.pValue <= fdr * (index + 1) / chiSqTestResult.length }
-          .map { case (_, index) => index }
-          .max
+        val maxIndex = if (selected.isEmpty) {
```

Comment:
How about replacing this new code with

```
if (selected.isEmpty) {
  Seq.empty
} else {
  val maxIndex = select.map { case (_, index) => index }.max
  tempRes.take(maxIndex + 1)
}
```

Might just be more straightforward and less change.

Reply:
sounds reasonable, I was already doubting whether doing what you proposed and what I did, so I am fine going that way. I am updating, thanks.